tests/aws/services/events/test_events_targets.py

```diff
@@ -37,6 +37,116 @@
+        snapshot.match("event_bus_response", event_bus_response)
+
+        # Step 2: Create a CloudWatch Logs log group
+        log_group_name = f"/aws/events/test-log-group-{short_uid()}"
+        aws_client.logs.create_log_group(logGroupName=log_group_name)
+        log_group_arn = f"arn:aws:logs:{region_name}:{account_id}:log-group:{log_group_name}"
+
+        # Step 3: Attach a resource policy to the log group to allow EventBridge to write to it
+        resource_policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Sid": "EventBridgePutLogEvents",
+                    "Effect": "Allow",
+                    "Principal": {"Service": "events.amazonaws.com"},
+                    "Action": ["logs:CreateLogStream", "logs:PutLogEvents"],
+                    "Resource": f"{log_group_arn}:*",
+                }
+            ],
+        }
+        policy_name = f"EventBridgePolicy-{short_uid()}"
+        aws_client.logs.put_resource_policy(
+            policyName=policy_name, policyDocument=json.dumps(resource_policy)
+        )
+
+        # Wait for the resource policy to take effect (only needed in AWS)
+        if is_aws_cloud():
+            time.sleep(10)
+
+        # Step 4: Create a rule on the event bus
+        rule_name = f"test-rule-{short_uid()}"
+        rule_response = events_put_rule(
+            Name=rule_name,
+            EventBusName=event_bus_name,
+            EventPattern=json.dumps(TEST_EVENT_PATTERN),
+        )
+        snapshot.match("rule_response", rule_response)
+
+        # Step 5: Add the CloudWatch Logs log group as a target
+        target_id = f"target-{short_uid()}"
+        put_targets_response = aws_client.events.put_targets(
+            Rule=rule_name,
+            EventBusName=event_bus_name,
+            Targets=[
+                {
+                    "Id": target_id,
+                    "Arn": log_group_arn,
+                }
+            ],
+        )
+        snapshot.match("put_targets_response", put_targets_response)
+        assert put_targets_response["FailedEntryCount"] == 0
+
+        # Step 6: Send an event to the event bus
+        event_entry = {
+            "EventBusName": event_bus_name,
+            "Source": TEST_EVENT_PATTERN["source"][0],
+            "DetailType": TEST_EVENT_PATTERN["detail-type"][0],
+            "Detail": json.dumps(EVENT_DETAIL),
+        }
+        put_events_response = aws_client.events.put_events(Entries=[event_entry])
+        snapshot.match("put_events_response", put_events_response)
+        assert put_events_response["FailedEntryCount"] == 0
+
+        # Step 7: Verify that the event was logged to the log group
+        def get_log_events():
+            response = aws_client.logs.describe_log_streams(logGroupName=log_group_name)
+            log_streams = response.get("logStreams", [])
+            if not log_streams:
+                raise Exception("No log streams found")
```

Comment:
nit: the common pattern when using the `retry` util function is normally using an `assert` lile:

```suggestion
            assert log_streams
```

could also optionally pass some info:
```suggestion
            assert log_streams, "No log streams found"
```